lib/buffer.js

```diff
@@ -505,7 +505,7 @@ Buffer.prototype.write = function(string, offset, length, encoding) {
 
   // XXX legacy write(string, encoding, offset, length) - remove in v0.13
   } else {
-    writeWarned = internalUtil.printDeprecationMessage(writeMsg, writeWarned);
+    writeWarned = internalUtil.deprecate(writeMsg, writeWarned);
```

Comment:
Actually, when `printDeprecationMessage` is invoked from a module (`sys` and `smalloc`) the actual caller will be at position 10 in the stack trace. But, in this case, the caller will be at position 2. When we invoke `printDeprecationMessage` from the `util.deprecate`, the actual caller will be at position 5. Now, we need to way to find out these three possible cases.